replication/binlogsyncer.go

```diff
@@ -424,6 +425,15 @@ func (b *BinlogSyncer) GetNextPosition() Position {
 	return b.nextPos
 }
 
+func (b *BinlogSyncer) checkFlavor() {
+	serverVersion := b.c.GetServerVersion()
+	if b.cfg.Flavor != MariaDBFlavor &&
+		strings.Contains(b.c.GetServerVersion(), "MariaDB") {
```

Comment:
The server returns `MariaDB` but our flavor requires lowercased, this is a bit uncomfortable. I orefer we let `ValidateFlavor` `ToLower` the string before compare.

Reply:
I changed `ValidateFlavor()` to do the `ToLower()` as suggested